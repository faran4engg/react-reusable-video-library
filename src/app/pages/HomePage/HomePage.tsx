import { FC } from 'react';
import { Helmet } from 'react-helmet-async';

import TrendingVideosContainer from 'app/domains/Videos/containers/TrendingVideos';
import { VideoGrid } from 'app/domains/Videos/components';

const HomePage: FC = () => (
  <>
    <Helmet>
      <title>Home | Islaamify360</title>
      <meta name="description" content="Home Page for Trending Videos" />
    </Helmet>

    <TrendingVideosContainer>
      {({ isLoading, videos }) => (
        <VideoGrid isLoading={isLoading} videos={videos} />
      )}
    </TrendingVideosContainer>
  </>
);

export default HomePage;
