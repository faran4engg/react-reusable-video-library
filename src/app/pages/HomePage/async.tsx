import { lazyLoad } from 'utils/loadable';
import { CardSkeleton } from 'app/domains/Common/components/loader';

export const HomePage = lazyLoad(
  () => import('./HomePage'),
  module => module.default,
  {
    fallback: (
      <div className="grid gap-6 mb-8 sm:grid-cols-2 md:grid-cols-3">
        <CardSkeleton noOfCards={3} />
      </div>
    ),
  },
);
