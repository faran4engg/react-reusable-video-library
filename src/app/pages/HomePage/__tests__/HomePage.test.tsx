import React from 'react';

import { create } from 'react-test-renderer';
import HomePage from '..';

jest.mock('react-query', () => ({
  useQuery: () => ({}),
}));

describe('<HomePage />', () => {
  const renderProps = jest.fn();

  const renderComponent = () =>
    create(
      <HomePage>
        {renderPropParams => {
          renderProps({ ...renderPropParams });
          return <div />;
        }}
      </HomePage>,
    );

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render and match the snapshot', () => {
    const component = renderComponent();

    expect(component).toMatchSnapshot();
  });

  it('should render with render prop', () => {
    const component = renderComponent();
    expect(component).toMatchSnapshot();
  });
});
