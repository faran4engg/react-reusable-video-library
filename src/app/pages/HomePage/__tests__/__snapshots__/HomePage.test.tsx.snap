// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render and match the snapshot 1`] = `
Array [
  "<Helmet />",
  <div
    className="text-gray-600 dark:text-gray-400"
  >
    <div
      className="grid gap-6 mb-8 sm:grid-cols-2 md:grid-cols-3"
    />
  </div>,
  <h1
    className="text-2xl"
  >
    clean code
  </h1>,
]
`;

exports[`<HomePage /> should render with render prop 1`] = `
Array [
  "<Helmet />",
  <div
    className="text-gray-600 dark:text-gray-400"
  >
    <div
      className="grid gap-6 mb-8 sm:grid-cols-2 md:grid-cols-3"
    />
  </div>,
  <h1
    className="text-2xl"
  >
    clean code
  </h1>,
]
`;
